packages/react-server-dom-webpack/src/ReactFlightWebpackNodeLoader.js

```diff
@@ -95,6 +95,106 @@ export async function getSource(
   return defaultGetSource(url, context, defaultGetSource);
 }
 
+function addLocalExportedNames(names: Map<string, string>, node: any) {
+  switch (node.type) {
+    case 'Identifier':
+      names.set(node.name, node.name);
+      return;
+    case 'ObjectPattern':
+      for (let i = 0; i < node.properties.length; i++)
+        addLocalExportedNames(names, node.properties[i]);
+      return;
+    case 'ArrayPattern':
+      for (let i = 0; i < node.elements.length; i++) {
+        const element = node.elements[i];
+        if (element) addLocalExportedNames(names, element);
+      }
+      return;
+    case 'Property':
+      addLocalExportedNames(names, node.value);
+      return;
+    case 'AssignmentPattern':
+      addLocalExportedNames(names, node.left);
+      return;
+    case 'RestElement':
+      addLocalExportedNames(names, node.argument);
+      return;
+    case 'ParenthesizedExpression':
+      addLocalExportedNames(names, node.expression);
+      return;
+  }
+}
+
+function transformServerModule(
+  source: string,
+  body: any,
+  url: string,
+  loader: LoadFunction,
+): string {
+  // If the same local name is exported more than once, we only need one of the names.
+  const localNames: Map<string, string> = new Map();
+  const localTypes: Map<string, string> = new Map();
+
+  for (let i = 0; i < body.length; i++) {
+    const node = body[i];
+    switch (node.type) {
+      case 'ExportAllDeclaration':
+        // If export * is used, the other file needs to explicitly opt into "use server" too.
+        break;
+      case 'ExportDefaultDeclaration':
+        if (node.declaration.type === 'Identifier') {
+          localNames.set(node.declaration.name, 'default');
+        } else if (node.declaration.type === 'FunctionDeclaration') {
+          if (node.declaration.id) {
+            localNames.set(node.declaration.id.name, 'default');
+            localTypes.set(node.declaration.id.name, 'function');
+          } else {
+            // TODO: This needs to be rewritten inline because it doesn't have a local name.
```

Comment:
makes sense, maybe could error for now so if someone discovers it they get an error message

Reply:
I'm hesitant to add any errors in this one since it's just a pass through and you can use it server->server. E.g. a nested object wouldn't error. Besides, nobody actually uses this.